validator/src/validator.rs

```diff
@@ -322,19 +338,9 @@ where
             .iter()
             .map(|validator| validator.voting_key.clone())
             .collect();
-        let key = self.voting_key();
         let network = Arc::clone(&self.network);
 
-        // TODO might better be done without the task.
```

Comment:
Is the "might better be done without the task." fixed by this commit? If not, it should probably stay.